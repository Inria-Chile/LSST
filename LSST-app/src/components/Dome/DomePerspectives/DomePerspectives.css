.dome-perspectives-container > .row {
    margin-bottom: 20px;
}

.dome-perspectives-bottom {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
}

.dome-perspectives-info {
    vertical-align: middle;
    display: inline-block;
    font-size: var(--dome-text-size);
}