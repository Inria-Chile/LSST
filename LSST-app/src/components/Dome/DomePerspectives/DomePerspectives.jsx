import React, { PureComponent } from 'react';
// import ReactDOM from 'react-dom';
import FrontView from './FrontView/FrontView';
import SideView from './SideView/SideView';
import Louvers from './Louvers/Louvers';
import './DomePerspectives.css';
import {DEMO_MODE} from '../../Utils/Utils';

class DomePerspectives extends PureComponent {

    constructor(props){
        super(props);
        this.state = {
            data: [],
            topWindScreenPos: 3.5,//min
            bottomWindScreenPos: 20,//min
        }
    }

    componentDidMount(prevProps, prevState){
        if(DEMO_MODE){
            setInterval( () => {
                this.props.updateShuttersAperture(11);
                let angle_0 = 35;
                let angle = Math.max(3.5, Math.ceil(Math.random()*(90-angle_0-10)));
                this.setState({
                    topWindScreenPos: angle,
                    bottomWindScreenPos: 90-(angle+angle_0),
                })
            }, 2000)
        }
    }

    componentWillReceiveProps(nextProps){
        if(this.props.telescopeElevation !== nextProps.telescopeElevation){
            this.setState({
                topWindScreenPos: Math.min(70, 90-(nextProps.telescopeElevation+11)).toFixed(2),//min
                bottomWindScreenPos: Math.max(5, Math.min(50, (nextProps.telescopeElevation-11)).toFixed(2)),//min
            })
        }
    }

    render() {
        let statusOpen = this.props.shuttersAperture > 0;
        return (
            <div className="dome-perspectives-container">
                <div className="row">
                    <div className="col-6">
                        <h5>Front</h5>
                        <FrontView width={500} height={300} id="shutters" shuttersAperture={this.props.shuttersAperture} updateShuttersAperture={this.props.updateShuttersAperture} 
                                topWindScreenPos={this.state.topWindScreenPos} bottomWindScreenPos={this.state.bottomWindScreenPos} scale={1.4} xOffset={0} yOffset={-0.0}/>
                    </div>
                    <div className="col-6">
                        <h5>Side</h5>
                        <SideView width={500} height={300} id="shutters" shuttersAperture={this.props.shuttersAperture} updateShuttersAperture={this.props.updateShuttersAperture} 
                                topWindScreenPos={this.state.topWindScreenPos} bottomWindScreenPos={this.state.bottomWindScreenPos} scale={1.4} xOffset={0} yOffset={-0.0}/>                    </div>
                </div>
                <div className="row dome-perspectives-bottom">
                    <div className="col-8">
                        <h5>Top</h5>
                        <Louvers width={500} height={500} id="louvers" scale={1.4} xOffset={-0.05} yOffset={0.15}/>
                    </div>
                    <div className="col-4 dome-perspectives-info">
                        <div>
                            Shutters status: <span className={"status-circle-"+(statusOpen ?'open':'closed')}> </span>
                                    <span className="dome-data">{this.props.shuttersAperture > 0 ? 'Open' : 'Closed'}</span>
                        </div>
                        <div>
                            Aperture: <span className="dome-data">{this.props.shuttersAperture} m </span>
                        </div>
                        <div>
                            Top windscreen position: <span className="dome-data">{this.state.topWindScreenPos} </span>
                        </div>
                        <div>
                            Bottom windscreen position: <span className="dome-data">{this.state.bottomWindScreenPos} </span>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default DomePerspectives;
