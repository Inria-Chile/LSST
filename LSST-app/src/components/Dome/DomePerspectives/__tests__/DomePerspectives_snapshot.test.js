import React from 'react';
import DomePerspectives from '../DomePerspectives';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from "enzyme"; 
import toJson from 'enzyme-to-json';
import SocketMock from 'socket.io-mock'
Enzyme.configure({ adapter: new Adapter() });


/*---- Snapshot testing ----*/
describe('DomePerspectives test', function(){
    const shuttersAperture = 10 ;
    const updateShuttersAperture = 10; 

    it('renders with shallow',()=>{
        let tree = shallow(<DomePerspectives shuttersAperture={shuttersAperture}
            updateShuttersAperture={updateShuttersAperture} />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});

