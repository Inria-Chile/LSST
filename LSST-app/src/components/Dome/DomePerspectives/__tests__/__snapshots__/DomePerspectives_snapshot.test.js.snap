// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomePerspectives test renders with new shallow 1`] = `
<div
  className="dome-perspectives-container"
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h5>
        Front
      </h5>
      <FrontView
        bottomWindScreenPos={20}
        height={300}
        id="shutters"
        scale={1.4}
        shuttersAperture={10}
        topWindScreenPos={3.5}
        updateShuttersAperture={10}
        width={500}
        xOffset={0}
        yOffset={-0}
      />
    </div>
    <div
      className="col-6"
    >
      <h5>
        Side
      </h5>
      <SideView
        bottomWindScreenPos={20}
        height={300}
        id="shutters"
        scale={1.4}
        shuttersAperture={10}
        topWindScreenPos={3.5}
        updateShuttersAperture={10}
        width={500}
        xOffset={0}
        yOffset={-0}
      />
                          
    </div>
  </div>
  <div
    className="row dome-perspectives-bottom"
  >
    <div
      className="col-8"
    >
      <h5>
        Top
      </h5>
      <Louvers
        height={500}
        id="louvers"
        scale={1.4}
        width={500}
        xOffset={-0.05}
        yOffset={0.15}
      />
    </div>
    <div
      className="col-4 dome-perspectives-info"
    >
      <div>
        Shutters status: 
        <span
          className="status-circle-open"
        >
           
        </span>
        <span
          className="dome-data"
        >
          Open
        </span>
      </div>
      <div>
        Aperture: 
        <span
          className="dome-data"
        >
          10
           m 
        </span>
      </div>
      <div>
        Top windscreen position: 
        <span
          className="dome-data"
        >
          3.5
           
        </span>
      </div>
      <div>
        Bottom windscreen position: 
        <span
          className="dome-data"
        >
          20
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DomePerspectives test renders with shallow 1`] = `
<div
  className="dome-perspectives-container"
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h5>
        Front
      </h5>
      <FrontView
        bottomWindScreenPos={20}
        height={300}
        id="shutters"
        scale={1.4}
        shuttersAperture={10}
        topWindScreenPos={3.5}
        updateShuttersAperture={10}
        width={500}
        xOffset={0}
        yOffset={-0}
      />
    </div>
    <div
      className="col-6"
    >
      <h5>
        Side
      </h5>
      <SideView
        bottomWindScreenPos={20}
        height={300}
        id="shutters"
        scale={1.4}
        shuttersAperture={10}
        topWindScreenPos={3.5}
        updateShuttersAperture={10}
        width={500}
        xOffset={0}
        yOffset={-0}
      />
                          
    </div>
  </div>
  <div
    className="row dome-perspectives-bottom"
  >
    <div
      className="col-8"
    >
      <h5>
        Top
      </h5>
      <Louvers
        height={500}
        id="louvers"
        scale={1.4}
        width={500}
        xOffset={-0.05}
        yOffset={0.15}
      />
    </div>
    <div
      className="col-4 dome-perspectives-info"
    >
      <div>
        Shutters status: 
        <span
          className="status-circle-open"
        >
           
        </span>
        <span
          className="dome-data"
        >
          Open
        </span>
      </div>
      <div>
        Aperture: 
        <span
          className="dome-data"
        >
          10
           m 
        </span>
      </div>
      <div>
        Top windscreen position: 
        <span
          className="dome-data"
        >
          3.5
           
        </span>
      </div>
      <div>
        Bottom windscreen position: 
        <span
          className="dome-data"
        >
          20
           
        </span>
      </div>
    </div>
  </div>
</div>
`;
